### DISCUSSION TITLE
What to expect from ConanCenter in the coming months!

### DISCUSSION THREAD

**Discussion by prince-chrismc** [conan-io/conan-center-index#19104] [Announcements]
With the Conan 2.0 release back in February, we aimed to have the top most downloaded recipes ready for Conan 2.0 with published packages. Today we have over 1123 recipes that have Conan 2.0 compatible packages, out of about 1296 that have already been migrated - with a further 153 pull requests open to migrate some of the remaining recipes to Conan 2.0 - **covering over 92% of the repository**. This wouldn’t have been possible without the help of our contributors! 👏 

So we wanted to update you on what we have been working on, and what changes to expect in the coming months:
- For Conan 2.0, packages for Visual Studio 2022 have been published and are now part of the CI checks for new Pull Requests
- We aim to have all the actively maintained recipes compatible with Conan 2.0 before the year. 
- New platforms will be added to the CI checks, including the most recent version of AppleClang on macOS. As previously announced, we will continue adding new platforms to the v2 CI pipeline, dedicating more resources to Conan 2.0 and reducing validated platforms for Conan 1.X. 

ℹ️ In order to continue our Conan 2.0 migration efforts, we plan to make _v2 compatibility mandatory for new PRs_ in Conan Center. **This will take effect on 29 August 2023.**

Exceptionally, Conan team maintainers may allow recipes that have not yet been migrated to be merged without supporting Conan 2.0. This continues from our recent changes of [deprecating the need for Conan 1.x-only test packages](https://github.com/conan-io/conan-center-index/discussions/17406) (e.g. `test_v1_package` folder) in ConanCenter.

At a future date, still to be determined, Conan Center recipes will no longer be compatible with Conan 1.X. We will provide details in the future to ensure that users who wish to stay on Conan 1.x a little longer are not disrupted by this change.

**Comment by jcar87** [comment]
🚀

**Comment by valgur** [comment]
I'm very excited to see CCI moving steadily closer to switching to v2 for good in the near future.

However, I have two related questions:

1. Does no longer requiring v1 compatibility also mean that v1 builds will be dropped entirely as well?
2. If so, are there any plans to update the CCI hooks and start applying them on the Conan v2 pipelines? Many of these checks have been extremely helpful in weeding out issues in recipes and build scripts (e.g. missing license, missing or incorrect headers, missing libraries, incorrect static/dynamic lib output, etc.).

**Reply by prince-chrismc** [reply]
> Does [that mean] v1 builds will be dropped entirely as well?

Yes! In fact that is the main intent. 

We do not want to see recipes intentionally being changed to break 1.x such as removing package info -- however using 2.0 exclusive features to improve the recipes will be. Hopefully the difference between those is clear

> Are there any plans to update the CCI hooks and start applying them on the Conan v2 pipelines?

Currently undecided. We are not happy with the current state of them and there are new members involved with new ideas and we have not come to a decision just yet. The focus is on improving the infrastructure, the master plan is if it's simple it should be more flexible. So how all this will fit in will likely come after :)

**Reply by 13steinj** [reply]
> We do not want to see recipes intentionally being changed to break 1.x such as removing package info -- however using 2.0 exclusive features to improve the recipes will be. Hopefully the difference between those is clear

Is there some strong backwards compatibility policy listed somewhere?

Some recipes obscure / hide flags / options that I'd love to be able to pass in, but changing the recipe to be more flexible would mean either breaking compatibility with older recipes (which historically, seems to not have been a problem, as I've repeatedly encountered such issues); or, doing it in a backwards compatible way that is subjectively ugly / a code-smell and objectively inconsistent with the rest of the recipe.

**Comment by uilianries** [comment]
We have been started the change:

- User feedback will be explicit about needing Conan 2.x success always: #19522
- The regression list for packages ported to Conan 2.x is no longer needed: #19523

We still need to update C3I Jenkins.
